frame/uniques/src/functions.rs

```diff
@@ -244,6 +314,21 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		Ok(())
 	}
 
+	/// Buy an item from a collection.
+	///
+	/// # Errors
+	/// This function returns an error in the following cases:
+	/// - The item or collection does not exist (`UnknownItem` or `UnknownCollection`).
+	/// - The buyer is the current owner of the item (`NoPermission`).
+	/// - The item is not for sale (`NotForSale`).
+	/// - The bid price is lower than the item's sale price (`BidTooLow`).
+	/// - The item is set to be sold only to a specific buyer, and the provided buyer is not the
+	///   whitelisted buyer (`NoPermission`).
+	/// - The currency transfer between the buyer and the owner fails for any reason.
+	///
+	/// # Returns
+	/// Returns `Ok(())` if the purchase is successful, otherwise, it returns an error indicating the
+	/// reason for failure.
```

Comment:
I think this is a bit superfluous, since the `Result` type already indicates that it could fail.

Reply:
Are you suggesting to remove the `# Result` section ? I'm open to it - I don't feel too strongly about keeping it. I put it in for completeness because it feels like what you'd expect after reading an `# Error` section.

Reply:
IMHO if there was something special about the Result; then yes. But just `Ok(())` is quite easy to understand.

Reply:
Yeah, I have the same thoughts as well, if it has the possibility of returning something other than the unit type then it should definitely be documented, otherwise I don't think it needs documenting.

Reply:
Sounds good, I'll update this PR to remove the Results section. Thanks !